src/sonic-ctrmgrd/ctrmgr/readiness_probe.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+# This script is used by k8s to check the readiness of containers
+# Check if the container is readiness or not, exit code 0 means readiness, others mean not readiness
+
+#### exit code contract, k8s only cares zero or not none-zero, but we want to use none-zero code to indicate different error
+# 0: readiness
+# 1: python script crach exit code
+# 2: supervisor start service doesn't exit normally
+# other exit code: returned by post_check_script, define in the post_check_script, should not include 1,2
+
+# check if the start service exists
+# if the start service exists, check if it exits normally
+# if the start service doesn't exist normally, exit with code 2
+pre_check_service_name="start"
+supervisorctl status |awk '{print $1}' |grep -w $pre_check_service_name > /dev/null
```

Comment:
>supervisorctl status [](http://example.com/codeflow?start=0&length=20)

You can use one command
```
supervisorctl status start
```

Reply:
Only do "supervisorctl status start", We can't do judgement by exit code, because start not existing and some failed state exit codes are the same. If only do "supervisorctl status start", need to judge by the outputs "start: ERROR (no such process)", "start  EXITED    Jun 21 05:28 PM". I do checking whether start exists in advance, I think code logic is more easy to understand here.

Reply:
This is an example
```admin@sonic:~$ docker exec -it lldp bash
root@sonic:/# supervisorctl status start
start                            EXITED    Jul 04 12:38 AM
root@sonic:/# supervisorctl status
dependent-startup                EXITED    Jul 04 12:38 AM
lldp-syncd                       RUNNING   pid 26, uptime 0:03:54
lldpd                            RUNNING   pid 20, uptime 0:03:57
lldpmgrd                         RUNNING   pid 30, uptime 0:03:52
rsyslogd                         RUNNING   pid 11, uptime 0:04:02
start                            EXITED    Jul 04 12:38 AM
supervisor-proc-exit-listener    RUNNING   pid 10, uptime 0:04:04
waitfor_lldp_ready               EXITED    Jul 04 12:38 AM

```

Reply:
Fixed